tests/test_wright_fisher.py

```diff
@@ -259,6 +265,31 @@ def test_with_mutations(self):
         for hap in ts.haplotypes():
             self.assertEqual(len(hap), ts.num_sites)
 
+    def test_with_recurrent_mutations(self):
+        # actually with only ONE site, at 0.0
+        N = 10
+        ngens = 100
+        tables = wf_sim(
+            N=N, ngens=ngens, mutation_rate=10.0, deep_history=False,
+            nloci=1, seed=self.random_seed)
+        self.assertEqual(tables.sites.num_rows, 1)
+        self.assertEqual(tables.sites.position, 0.0)
+        self.assertGreater(tables.mutations.num_rows, 0)
```

Comment:
We should make sure running ``haplotypes`` succeeds here as well, i.e., load the pre-simplify tables into a tree sequence, and get the haplotypes.

Reply:
It is [already being tested](https://github.com/jeromekelleher/msprime/blob/master/tests/test_wright_fisher.py#L364) that this works.

Strangely!  This should *not* work, because in laying down mutations I [don't check the ancestral state](https://github.com/jeromekelleher/msprime/blob/master/tests/test_wright_fisher.py#L117).

Reply:
That's why we should run it, to make sure that the pre-simplify mutations make sense. We're not testing 'haplotypes' here, rather we're checking that the input data makes sense.

Reply:
Yes, I agree. I did insert code to make sure that haplotypes ran, and they did; but removed it when I noticed it was being tested elsewhere.  With correct links: [here](https://github.com/petrelharp/msprime/blob/mut_parent/tests/test_wright_fisher.py#L328) we run WF sims with three loci and mutation rate = 5.0, so are getting multiple muts per site.

So, I claim it *does* work, but is almost certainly violating the parent != child mutation condition?  I have not verified that, so maybe something is not as I expect.  (Later: now, am translating kidz -> school.)

Reply:
I've added in parent mutation to the WF sim and am now [loading and checking haps](https://github.com/petrelharp/msprime/blob/7b7983d58fa8773f4e927f11945405ab0dd3b617/tests/test_wright_fisher.py#L302) and no complaints.  I've checked, and in this WF sim there are *lots* of consecutive mutations with the same state, e.g.
```
id      site    node    derived_state   parent
0       0       10      A       -1
1       0       10      T       0
2       0       10      C       1
3       0       10      T       2
4       0       10      T       3
```

I think that maybe you are not actually checking for the parent != child mutation property?